Text-Extraction Libraries Project Proposal

kalpitk opened the pull request.

kalpitk committed:
Add first version of proposal

jaylett commented on proposal.rst:
```diff
@@ -184,14 +206,18 @@ Motivations
 .. do and how your project will improve things for them.
 
-FILLME
+This project removes dependencies on external software, adds support where it
```
You don't mention in your proposal where you'd *remove* dependencies on external software, and the project outline we provide doesn't mention it either. You might change some dependencies for others, is that what you mean? It's worth taking the time to be as clear as possible in your proposal.

jaylett commented on proposal.rst:
```diff
@@ -271,7 +309,34 @@ Project Timeline
+    * Read documentation relating to the required libraries.
+
+First Coding Period (June 1 - June 28)
+    * Implement first handler, update build system, and add new worker to omindex. (7 days)
```
What is your first handler going to be? Are you thinking libarchive? What file formats would that support? libarchive is unlikely to directly support file formats; you're probably going to have to do some more work, so if that's your preferred first subproject then you should think about the different stages you'll go through to add support for one or more file formats, using libarchive to unpack the zip or similar that the files are stored in.

kalpitk replied:
ODT (OpenDocument Text) with help of libarchive will be my first target.

jaylett commented on proposal.rst:
```diff
@@ -271,7 +309,34 @@ Project Timeline
+    * Implement first handler, update build system, and add new worker to omindex. (7 days)
+    * Look for issues, take feedback, improve performance. (6 days)
+    * Document the code. (4 days)
+    * Finalise the PR and merge. (2 days)
```
From experience, your first PR of GSoC will almost certainly take longer than that, particularly if you haven't had a PR merged before or during the community bonding period. You can (and should) overlap dealing with rounds of feedback on your first PR with later work, but the priority should always be getting the PR merged (otherwise you can end up with multiple open PRs, none ready to merge).

kalpitk replied:
Thanks for the feedback. The time taken in feedback and resolving issues is the effective time. It will happen concurrently with the next task.

jaylett commented on proposal.rst:
```diff
@@ -271,7 +312,34 @@ Project Timeline
+    * Finalise the PR and merge. (2 days)
+
+Second Coding Period (July 6 - July 27)
+    For two types:
```
What types are you proposing? It's fine to suggest something here but change your mind after later discussion. But having something concrete laid out in your proposal allows you to think about the specifics of the work needed. "Add support for a new type" is very abstract. Are you going to add support for a single file format using a specialist library (perhaps from the Document Liberation Project)? Or are you going to find another library that could support several different file formats at once (which might take more work)?

kalpitk replied:
I am proposing work on OpenDocument formats only. Updated the proposal.

jaylett commented on proposal.rst:
```diff
@@ -271,7 +312,34 @@ Project Timeline
+First Coding Period (June 1 - June 28)
+    * Implement first handler, update build system, and add new worker to omindex. (7 days)
+    * Look for issues, take feedback, improve performance. (6 days)
+    * Document the code. (4 days)
```
What sort of documentation are you planning? 4 days seems a long time if you are adding support for one or two file formats. If you flesh out the detail of what support you're adding in this subproject, it may become clearer what documentation you need.

kalpitk replied:
Documenting the code + Document on how to add support for other zip based xml formats. Updated the proposal to reflect this.

jaylett reviewed (commented):
I think you need to be more specific about what your project is going to do, in terms of which libraries and which file formats.

You could also look at [the open PR](https://github.com/xapian/xapian/pull/280) that was left from Bruno's project, and determine if this needs finishing off and merging, and if so whether that's at the start of the project or later on.

ojwb commented:
Note you'll need to submit a "final" PDF version of this at https://summerofcode.withgoogle.com/ to be eligible for GSoC (and I don't seem to be able to see one currently, though possibly I'm failing to match github user name and real name).

You should be able to generate a suitable pdf from the .rst version using `make pdf`.

You can update the "final" version as many times as you like up to the deadline, so it's wise to submit early and not run the risk of internet problems preventing you submitting in time.

kalpitk committed:
Update the proposal
- List the probable extensions whose support will be added
- Reworded the proposal
- Add Pull Request 'Add omindextest to Omega' to the plan

kalpitk commented:
> I think you need to be more specific about what your project is going to do, in terms of which libraries and which file formats.
> 
> You could also look at [the open PR](https://github.com/xapian/xapian/pull/280) that was left from Bruno's project, and determine if this needs finishing off and merging, and if so whether that's at the start of the project or later on.

Included in the proposal.
I am proposing work on archive based xml formats only along with the above PR.
If time permits, we may think of one-two other formats.

kalpitk commented:
> Note you'll need to submit a "final" PDF version of this at https://summerofcode.withgoogle.com/ to be eligible for GSoC (and I don't seem to be able to see one currently, though possibly I'm failing to match github user name and real name).
> 
> You should be able to generate a suitable pdf from the .rst version using `make pdf`.
> 
> You can update the "final" version as many times as you like up to the deadline, so it's wise to submit early and not run the risk of internet problems preventing you submitting in time.

I just uploaded the current version of proposal on GSoC website.

ojwb commented:
> I just uploaded the current version of proposal on GSoC website.

I don't see it in the list of proposals submitted to Xapian...

kalpitk commented:
> > I just uploaded the current version of proposal on GSoC website.
> 
> I don't see it in the list of proposals submitted to Xapian...

Title of Proposal is "Text-Extraction Libraries", my name is "Kalpit Kothari". Please verify.

ojwb commented:
> Title of Proposal is "Text-Extraction Libraries", my name is "Kalpit Kothari". Please verify.

It's definitely not in our proposal list.  Did you submit it to a different org by mistake perhaps?

ojwb commented:
Thanks for you proposal.

The purpose of this PR was to provide feedback on the draft proposal rather than actually as something to merge into the template, so closing as the proposal deadline has now passed.
